tree/dataframe/test/dataframe_helpers.cxx

```diff
@@ -324,6 +324,88 @@ TEST(RDFHelpers, SaveGraphSharedDefines)
    EXPECT_EQ(graph, expected);
 }
 
+TEST(RDFHelpers, GraphAsymmErrorsContainers)
+{
+   const std::vector<double> xx = {-0.22, 0.05, 0.25, 0.35, 0.5, 0.61, 0.7, 0.85, 0.89, 0.95};
+   const std::vector<double> yy = {1., 2.9, 5.6, 7.4, 9, 9.6, 8.7, 6.3, 4.5, 1};
+   const std::vector<double> exl = {.05, .1, .07, .07, .04, .05, .06, .07, .08, .05};
+   const std::vector<double> exh = {.02, .08, .05, .05, .03, .03, .04, .05, .06, .03};
+   const std::vector<double> eyl = {.8, .7, .6, .5, .4, .4, .5, .6, .7, .8};
+   const std::vector<double> eyh = {.6, .5, .4, .3, .2, .2, .3, .4, .5, .6};
+
+   auto df = RDataFrame(1)
+                .Define("xx", [&] { return xx; })
+                .Define("yy", [&] { return yy; })
+                .Define("exl", [&] { return exl; })
+                .Define("exh", [&] { return exh; })
+                .Define("eyl", [&] { return eyl; })
+                .Define("eyh", [&] { return eyh; });
+
+   auto gr1 = df.GraphAsymmErrors("xx", "yy", "exl", "exh", "eyl", "eyh");
+
+   for (size_t i = 0; i < xx.size(); ++i) {
+      EXPECT_DOUBLE_EQ(gr1->GetX()[i], xx[i]);
+      EXPECT_DOUBLE_EQ(gr1->GetY()[i], yy[i]);
+      EXPECT_DOUBLE_EQ(gr1->GetEXlow()[i], exl[i]);
+      EXPECT_DOUBLE_EQ(gr1->GetEXhigh()[i], exh[i]);
+      EXPECT_DOUBLE_EQ(gr1->GetEYlow()[i], eyl[i]);
+      EXPECT_DOUBLE_EQ(gr1->GetEYhigh()[i], eyh[i]);
+   }
+}
+
+TEST(RDFHelpers, GraphAsymmErrorsScalars)
+{
+   double yy = 3.4;
+   double exl = .05;
+   double exh = .02;
+   double eyl = .8;
+   double eyh = .6;
+
+   auto df = RDataFrame(2)
+                .DefineSlotEntry("xx", [](unsigned int, ULong64_t xx) { return xx; })
+                .Define("yy", [&] { return yy; })
+                .Define("exl", [&] { return exl; })
+                .Define("exh", [&] { return exh; })
+                .Define("eyl", [&] { return eyl; })
+                .Define("eyh", [&] { return eyh; });
```

Comment:
it would be better to return different values from different entries, e.g. with [DefineSlotEntry](https://root.cern/doc/master/classROOT_1_1RDF_1_1RInterface.html#acf8eed4c366d6cedb2acfcb96bd12a5f)

Reply:
Noted, but how exactly? I would have to cast `ULong64_t` to `double`, which looks "dangerous"

Reply:
it's not dangerous, at least in this context. You'd just get a double precision representation of `0`, `1`, `2` etc.

but actually what I was suggesting was:

```suggestion
                .DefineSlotEntry("eyh", [&](unsigned int, ULong64_t e) { return eyh + e; });
```